### ISSUE TITLE
[Bug] Rendering stops working for some zoom-levels when map-width is a power-of-two number

### ISSUE THREAD

**Issue by usefulthink** [visgl/deck.gl#6901]
### Description

[Live Example](https://storage.ubidev.net/deckgl-power-of-two-width-bug/index.html)

For combinations of zoom/map-size where the width of the maptiles in pixels
matches the width of the map in the dom, rendering stops working.

Example:

- a map at zoom-level 3 wont render when map-size is 2048px
- at zoom-level 2, rendering fails for a width of 1024px
- same for zoom 4 / 4096px width and zoom 1 / 512px width

Moving the map around will sometimes result in a working rendering.

### Flavors

- [ ] React
- [ ] Python/Jupyter notebook
- [ ] MapboxLayer
- [X] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] DeckLayer/DeckRenderer for ArcGIS

### Steps to Reproduce

Visit the Live Example and press one of the buttons: https://storage.ubidev.net/deckgl-power-of-two-width-bug/index.html

(sourcecode for the example can be found here: https://github.com/ubilabs/deckgl-power-of-two-width-bug)

### Environment

- Framework version: deck.gl/google-maps@8.7.9
- Browser: all affected (tested in latest FF/Chrome/Safari)
- OS: OSX 10.15

**Comment by usefulthink** [comment]
A workaround that worked for us is to add css to make sure that the map can never have an affected size: 

```css
@media (width: 512px), (width: 1024px), (width: 2048px), (width: 4096px) {
  #map {
    width: calc(100vw + 1px);
  }
}
```

**Comment by Pessimistress** [comment]
This should be fixed by v8.7.10.
